File: backend/app/schema/listing.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class ApartmentListing(BaseModel):
    id: str
    listing_date: datetime
    rent: float
    broker_fee: float
    square_footage: float
    bedrooms: int
    bathrooms: int
    available_date: datetime
    street_address: str
    zillow_url: str
    created_at: datetime
    updated_at: datetime

File: frontend/src/schema/listing.ts
import { z } from 'zod';

export const ListingSchema = z.object({
  id: z.string().uuid(),
  listingDate: z.date(),
  rent: z.number().positive(),
  brokerFee: z.number().nonnegative(),
  squareFootage: z.number().positive(),
  bedrooms: z.number().int().positive(),
  bathrooms: z.number().int().positive(),
  availableDate: z.date(),
  streetAddress: z.string(),
  zillowUrl: z.string().url(),
  createdAt: z.date(),
  updatedAt: z.date(),
});

export type Listing = z.infer<typeof ListingSchema>;